src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -183,6 +183,9 @@ internal void SubscribeExternalErrorDiagnosticUpdateSourceToSolutionBuildEvents(
 
     public async Task InitializeUIAffinitizedServicesAsync(IAsyncServiceProvider asyncServiceProvider)
     {
+        // Yield the thread, so the caller can proceed and return immediately.
+        await Task.Yield();
+
         // Create services that are bound to the UI thread
         await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_threadingContext.DisposalToken);
```

Comment:
>SwitchToMainThreadAsync [](http://example.com/codeflow?start=52&length=23)

isn't there an always yield param to this?

Reply:
yup.  thanks.